### PR TITLE
[SERVICE-416] Switch Monitor Scaling Dialog to an OpenFin toast Notification

### PR DESCRIPTION
**Pull request by ryanpetersonOF**
**Do Not Merge Yet!!!** 

Switches the previous scaling error window to be a notification.  The previous was handy, so I have worked around it and added the ability to spawn in a window or in a notification.

We are still waiting on further design and/or wording updates, so no merging yet!

### CODE REVIEW

File: res/provider/errors/error.html [hunk 1]
```diff
@@ -95,19 +146,21 @@
...
 			const wnd = await fin.Window.getCurrent();
-			const { customData: { title, error } } = await wnd.getOptions();
-			const errorTitleEl = document.querySelector('#error-title');
-			const errorMessageEl = document.querySelector('#error-message');
-			
-			if (title) {
-				errorTitleEl.innerHTML = title;
-			} else {
-				errorTitleEl.remove();
+			const { customData } = await wnd.getOptions();
```
Comment (bryangaleOF) [inline res/provider/errors/error.html]: Spacing consistency

File: res/provider/errors/error.html [hunk 2]
```diff
@@ -125,6 +178,21 @@
...
 
+		// Writes error and title to DOM
+		function writeError(title, error, icon) {
+			if (title) {
+				errorTitleEl.innerHTML = title;
+			} else {
+				errorTitleEl.remove();
+			}
+			
+			if(icon) {
```
Comment (bryangaleOF) [inline res/provider/errors/error.html]: Spacing consistency

File: src/provider/utils/error.ts [hunk 3]
```diff
@@ -1,33 +1,87 @@
...
+ * Creates an error notice in the form of a openfin window or notification.
+ * @param title Title of the error
+ * @param message Error message
+ * @param icon Error message icon.  Default is red error cross.
+ * @param showOnceKey This key is used to identify if the error has occurred.  If set, this error will only show once and never again.
+ * @param medium Notification or Window
+ */
+export async function createErrorNotice(title: string, message: string, icon: string|null, showOnceKey: string|null, medium: ErrorMedium) {
+    if (!showOnceKey || !localStorage.getItem(showOnceKey)) {
+        const launchDir = location.href.slice(0, location.href.lastIndexOf('/'));
```
Comment (bryangaleOF) [inline src/provider/utils/error.ts]: We've got this exact line 6 places in layouts. Wondering if it's worth pulling out somewhere

File: res/provider/errors/error.html [hunk 4]
```diff
@@ -84,8 +120,23 @@
...
 	<script>
-		document.addEventListener('DOMContentLoaded', getExceptionData);
+		const errorTitleEl = document.querySelector('#error-title');
+		const errorMessageEl = document.querySelector('#error-message');
+		const errorIconEl = document.querySelector("#error-icon");
+
+		document.addEventListener('DOMContentLoaded', onWindowLoad);
 
+		/**
+		 * This is the hardcoded openfin Notification method to retrieve a notification method.  This will never be triggerd when viewed in a window.
```
Comment (MichaelMCoates) [inline res/provider/errors/error.html]: *triggered

File: src/provider/utils/error.ts (hunk 3 above)
Comment (MichaelMCoates) [reply src/provider/utils/error.ts]: 👍

**Comment by MichaelMCoates** [comment]
Question for Product: Do we only want the scaling notification to show once? 

Otherwise, looks and works great.

File: src/provider/utils/error.ts (hunk 3 above)
Comment (ryanpetersonOF) [reply src/provider/utils/error.ts]: Wow, definitely worth it

**Comment by ryanpetersonOF** [comment]
@MichaelMCoates I'm in talks with them about wording on the notice.  I'll follow up with an update on the once stuff as well

File: res/provider/errors/error.html (hunk 4 above)
Comment (ryanpetersonOF) [reply res/provider/errors/error.html]: 👍

File: res/provider/errors/error.html (hunk 1 above)
Comment (ryanpetersonOF) [reply res/provider/errors/error.html]: 👍

File: res/provider/errors/error.html (hunk 2 above)
Comment (ryanpetersonOF) [reply res/provider/errors/error.html]: 👍

File: src/provider/utils/error.ts (hunk 3 above)
Comment (tomrobertsOF) [reply src/provider/utils/error.ts]: Make a node module out of it: 'get-relative-url' :P

Comment (ryanpetersonOF) [reply src/provider/utils/error.ts]: @tomrobertsOF Why not just use jests `getRealPath` method that works so great.  Jokes aside, I have made a util function for this: `getHrefDirectory()` and added it everywhere that had this line.

**Comment by pjbroadbent** [comment]
Change of approach - PR replaced by #366 instead.

**Pull request closed without merging**